Portal/View/Components/Progress.swift

```diff
@@ -6,16 +6,18 @@
 //  Copyright © 2017 Guido Marucci Blas. All rights reserved.
 //
 
-public struct ProgressCounter {
+public struct ProgressCounter: AutoPropertyDiffable {
     
-    public static let initial = ProgressCounter()
+    internal static let initial = ProgressCounter()
+    
+    private var partial: UInt
```

Comment:
Why did you make this variable private?

Reply:
I didn't know what to do here, if make all the unnecessary properties (for the user) privates or let them public but the apply method would look something like this:

``` swift code
fileprivate func apply(changeSet: [ProgressCounter.Property]) {
        for property in changeSet {
            switch property {
                
            case .progress(let progress):
                self.progress = progress

            case .partial(_):
                   break

            case .total(_):
                   break

            case .remaining(_):
                   break
            }
        }
    }
```
because the only ways to modify a ProgressCounter is creating a new one with `init?(partial: UInt, total: UInt)` or use add progress function `add(progress: UInt) -> ProgressCounter?`.

Reply:
Oh I see. There are 2 options. Make it private or avoid them while generating the `ProgressCounter.Property` enum object. You can do that avoid adding a comment in the code like https://github.com/guidomb/Portal/blob/view-diff/Portal/View/Components/Button.swift#L17. We should add a new annotation, something like `// sourcery: ignoreInChangeSet` and then modify the template that generates the code in https://github.com/guidomb/Portal/blob/view-diff/Portal/Templates/AutoPropertyDiffable.stencil#L7 by adding something like

```
{% if not instanceVariable.annotations.ignoreInChageSet %}
 REST OF THE CODE
{% endif %}
```

Reply:
If we do this we should also modify the chageset generation to avoid comparing properties that are ignored. Talk with @pablog so he can help you with this